mautic/mautic
# Removing the fingerprint tracking

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | /
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | Y
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

The fingerprint tracking was causing only problems. Some devices like iPhones had the same fingerprint so it sent all page hits into one contact. Basically, this feature is a trap for users who doesn't know the problems it can cause when they turn it on.

The Fingerprint JS library enhanced the tracking for other information. I managed to replace it with vanilla JS like this:
```js
timezone_offset: new Date().getTimezoneOffset(),
resolution: window.screen.width + 'x' + window.screen.height,
platform: m.getOs(),
do_not_track: navigator.doNotTrack == 1
```
The only param I was not able to (or willing to) implement was `adblock` which require a bogus request to check if an adblocker stop it or not. It's not worth it.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. No need to replicate the issue. It has been reported multiple times.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Ensure the page tracking via tracking JS still works with all the details as before. Like resolution or timezone offset.

#### List backwards compatibility breaks:
1. Check the changes in the UPGRADE-3.0.md file
